### DISCUSSION TITLE
Run Vitest Programmically

### DISCUSSION THREAD

**Discussion by ScottAwesome** [vitest-dev/vitest#1053] [Ideas]
I have a (hopefully) simple idea that I think would take Vitest to the next level.

My big idea here would be for a Vitest API to run tests in a programmatic fashion. This would allow things such as

- Being able to selectively run tests based on say, environment variables. For instance, I may want to run my node tooling tests with a different set of defaults over my browser packages, for example. Or I may simply want to chunk how I run them across my repository (particularly if a have a large monorepo)

- Allow for user land tooling. For instance, if I'm integrating Vitest into a CLI, how do I do that today? Do I have to pass arguments to the Vitest executable? That means I'm managing things like `bin` file locations, crawling of the filesystem etc. and having to deal with process management

- Allow for other tools to handle meta configuration attributes of Vitest in an easier fashion

This would make it significantly easier to integrate Vitest into a variety of existing meta tooling, like `nx`, `rush`, `heft` and other coordinated task / runner scenarios.

**Comment by webistomin** [comment]
Here is [the answer](https://github.com/vitest-dev/vitest/issues/2167)
